File: TYPESCRIPT PROGRAMMING/Test77.ts
// 77. HCF AND LCM OF TWO NUMBERS
const HcfLcm = (num1: number, num2: number): void => {
    let a: number = num1, b: number = num2;
    let t: number = 0, HCF: number = 0, LCM: number = 0;
    while(b !== 0) {
        t = b;
        b = a % b;
        a = t;
    }
    HCF = a;
    LCM = (num1 * num2) / HCF;
    console.log(`LCM: ${LCM}`);
    console.log(`HCF: ${HCF}`);
}
HcfLcm(12, 15);
HcfLcm(120, 160);

File: JAVA PROGRAMMING/Test77.java
// 77. HCF AND LCM OF TWO NUMBERS
import java.util.Scanner;
public class Test {
    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
		System.out.println("Enter Num1 : ");
		int num1 = in.nextInt();
		System.out.println("Enter Num2 : ");
		int num2 = in.nextInt();

		int a = num1, b = num2, t, HCF, LCM;
		while(b != 0)	{
			t = b;
			b = a % b;
			a = t;
		}
		HCF = a;
		LCM = (num1 * num2) / HCF;

		System.out.println("LCM : "+LCM);
		System.out.println("HCF : "+HCF);
        in.close();
    }
}